chain/bitcoind_conn.go

```diff
@@ -114,6 +133,28 @@ type BitcoindConn struct {
 	// events.
 	zmqTxConn *gozmq.Conn
 
+	// rpcPolling sets whether we're polling RPC for block and tx events,
+	// instead of using ZMQ.
+	rpcPolling bool
+
+	// mempool keeps track of all of the transactions, needed
+	// when polling rpc to obtain the latest transactions. This map allows
+	// ensure we aren't processing any transactions more than
+	// once.
+	//
+	// We'll keep a timestamp of when we retrieve each transaction, so
+	// that we can prune any transactions that are too old. This helps to
+	// prevent against forever holding transactions that were added to the
+	// mempool, but were never confirmed.
+	mempoolMtx sync.Mutex
+	mempool    map[chainhash.Hash]time.Time
```

Comment:
where does this pruning happen?

Reply:
That piece is in this commit https://github.com/btcsuite/btcwallet/pull/784/commits/97743bb76f52903418f44696bf3784d8b1e357c2